src/server/qgshttprequesthandler.cpp

```diff
@@ -218,21 +218,10 @@ void QgsHttpRequestHandler::sendResponse()
   clearBody();
 }
 
-QByteArray QgsHttpRequestHandler::getResponse( const bool returnHeaders,
-    const bool returnBody )
+QPair<QByteArray, QByteArray> QgsHttpRequestHandler::getResponse()
 {
-  if ( ! returnHeaders )
-  {
-    return mResponseBody;
-  }
-  else if ( ! returnBody )
-  {
-    return mResponseHeader;
-  }
-  else
-  {
-    return mResponseHeader.append( mResponseBody );
-  }
+  QPair<QByteArray, QByteArray> response( mResponseHeader, mResponseBody );
+  return response;
```

Comment:
Thanks @m-kuhn ! 
Yes, the QgsHttpResponse is the way to go and it is in my TODO list since Nodebo but it requires much more work, for the time being I'm happy with this small improvement.
Also note that this is all about plugins, the standard fcgi server prints to stdout immediately and doesn't accumulate headers and body into QByteArray(s).